src/rgw/rgw_amqp.cc

```diff
@@ -53,14 +53,29 @@ static const int RGW_AMQP_NO_REPLY_CODE =                 0x0;
 
 // the amqp_connection_info struct does not hold any memory and just points to the URL string
 // so, strings are copied into connection_id_t
-connection_id_t::connection_id_t(const amqp_connection_info& info, const std::string& _exchange)
-    : host(info.host), port(info.port), vhost(info.vhost), exchange(_exchange), ssl(info.ssl) {}
+connection_id_t::connection_id_t(
```

Comment:
why changing the amqp code?

Reply:
coz amqp too faces same issue, if for some reason there is typo in the username and password, coz the connection is pooled, the new connection never uses the new userid and passwd

Reply:
amqp code needs a major rewrite (there are several issues fixed for kafka that still exist in amqp).
also, we would need to test that.
would recommend splitting amqp work to another PR

Reply:
i am afraid this will be forgotten while we refactor the amqp
since we are doing this for kafka, why not just do it for amqp as well, coz amqp also faces the same issue ? and its similar code changes so its logical to be part of same commit ?

Reply:
agree that this is needed there. but there are many other (critical) fixes we did on the kafka code that are needed there as well.
main reason that i want it in a separate PR is testing.

Reply:
done, removed amqp code.